wagtail/wagtailsearch/signal_handlers.py

```diff
@@ -22,16 +26,24 @@ def post_save_signal_handler(instance, **kwargs):
     indexed_instance = get_indexed_instance(instance)
 
     if indexed_instance:
-        for backend in get_search_backends(with_auto_update=True):
-            backend.add(indexed_instance)
+        for backend_name, backend in get_search_backends_with_name(with_auto_update=True):
+            try:
+                backend.add(indexed_instance)
+            except:
+                # Catch and log all errors
+                logger.exception("Exception raised while adding an object into the '%s' search backend", backend_name)
```

Comment:
Can the class and primary key, and maybe even the string representation of the model instance be included in this error? It would help debugging if a single type of model had an issue with its search config and kept failing.